kubernetes/Upgrading_CoreDNS.md

```diff
@@ -0,0 +1,197 @@
+
+# Handling CoreDNS Upgrades in Kubernetes Clusters
+
+When upgrading a Kubernetes cluster, some care should be taken when upgrading the CoreDNS component to avoid backward incompatible configuration failures. At a high level, you'll want to review the CoreDNS release notes to see if the new version introduces backward incompatibilities or deprecations and adjust your Corefile as needed before upgrading CoreDNS.
+
+
+## Identifying Backward Incompatibilities and Deprecations
+
+To identify possible backward incompatibilities, you'll need to review the CoreDNS release notes.  CoreDNS Release Notes are located in the [CoreDNS blog](https://coredns.io/blog/). The CoreDNS deprecation policy is such that we will only introduce backward incompatibilities in x.x.0 and x.x.1 releases.  So, for example, if you are upgrading from 1.1.5, to 1.3.1, you should check release notes for 1.2.0, 1.2.1, 1.3.0, and 1.3.1 for any deprecation/backward incompatibility notices.
+
+If you’ve discovered any backward incompatibility notices, you should review your Corefile to see if you are affected.
+
+
```

Comment:
Should we add a paragraph to show how to make it when downtime is not so big problem:
- upgrade your deployment manually ? or through tool ?
- edit configmap to tune the config
- verify coredns starts with checking pods and logs

Reply:
@fturib, adjusted per your comments, PTAL.